File: src/main/java/uk/gov/justice/digital/hmpps/riskprofiler/datasourcemodel/Viper.java
package uk.gov.justice.digital.hmpps.riskprofiler.datasourcemodel;

import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import java.math.BigDecimal;

@Builder
@Data
@EqualsAndHashCode(of = { "nomisId"})
@ToString
public class Viper implements RiskDataSet {
    public static int NOMIS_ID_POSITION = 1;
    public static int SCORE_POSITION = 3;

    private String nomisId;
    private BigDecimal score;

    public String getKey() {
        return nomisId;
    }

}

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/datasourcemodel/Viper.kt
package uk.gov.justice.digital.hmpps.riskprofiler.datasourcemodel

import java.math.BigDecimal

data class Viper(private val nomisId: String) : RiskDataSet {
  override fun getKey() = nomisId
  var score: BigDecimal? = null

  companion object {
    @JvmField
    var RECORD_ID = 0
    @JvmField
    var NOMIS_ID_POSITION = 1
    @JvmField
    var SCORE_POSITION = 7
  }
}
